python_modules/dagster/dagster/_core/execution/plan/execute_plan.py

```diff
@@ -35,106 +35,132 @@ def inner_plan_execution_iterator(
-
-            # pass a list of step events to hooks
-            for hook_event in _trigger_hook(step_context, step_event_list):
-                yield hook_event
+                            log_capture_error = e
+
+                        if not log_capture_error:
+                            yield DagsterEvent.legacy_compute_log_step_event(step_context)
+
+                        for step_event in check.generator(
+                            dagster_event_sequence_for_step(step_context)
+                        ):
+                            check.inst(step_event, DagsterEvent)
+                            step_event_list.append(step_event)
+                            yield step_event
+                            active_execution.handle_event(step_event)
+
+                        active_execution.verify_complete(pipeline_context, step.key)
+
+                        try:
+                            step_stack.close()
+                        except Exception:
+                            yield DagsterEvent.engine_event(
+                                plan_context=step_context,
+                                message="Exception while cleaning up compute log capture",
+                                event_specific_data=EngineEventData(
+                                    error=serializable_error_info_from_exc_info(sys.exc_info())
+                                ),
+                            )
+                    else:
+                        # we have already set up the log capture at the process level, just handle the
+                        # step events
+                        for step_event in check.generator(
+                            dagster_event_sequence_for_step(step_context)
+                        ):
+                            check.inst(step_event, DagsterEvent)
+                            step_event_list.append(step_event)
+                            yield step_event
+                            active_execution.handle_event(step_event)
+
+                        active_execution.verify_complete(pipeline_context, step.key)
+
+                # process skips from failures or uncovered inputs
+                for event in active_execution.plan_events_iterator(pipeline_context):
+                    step_event_list.append(event)
+                    yield event
+
+                # pass a list of step events to hooks
+                for hook_event in _trigger_hook(step_context, step_event_list):
+                    yield hook_event
+
+            try:
+                capture_stack.close()
+            except Exception:
+                yield DagsterEvent.engine_event(
+                    plan_context=pipeline_context,
+                    message="Exception while cleaning up compute log capture",
+                    event_specific_data=EngineEventData(
+                        error=serializable_error_info_from_exc_info(sys.exc_info())
+                    ),
+                )
```

Comment:
i dont know what it looks like - but there has to be a better way to structure this code this is really gnarly